R/utils.R

```diff
@@ -60,6 +60,46 @@ find_phantom <- function() {
   path.expand(path)
 }
 
+phantomjs_cmd_result = function(args, wait = TRUE) {
+  # Retrieve and store output from STDOUT
+  utils::capture.output(invisible(phantom_run(args = args, wait = wait)),
+                        type = "output")
+}
+
+phantomjs_version = function() {
+  phantomjs_cmd_result("--version")
+}
+
+#' Determine if PhantomJS is Installed
+#'
+#' Verifies that a version of PhantomJS is installed and available for use
+#' on the user's computer.
+#'
+#' @return \code{TRUE} if the PhantomJS is installed. Otherwise, \code{FALSE}
+#' if PhantomJS is not installed.
+#'
+#' @export
+is_phantomjs_installed <- function() {
+  suppressMessages( !is.null(find_phantom()) )
+}
+
+is_phantomjs_version_latest = function(requested_version) {
+  # Ensure phantomjs is installed if not, request an update
+  if (!is_phantomjs_installed()) {
+    return(FALSE)
+  }
+
+  # Obtain the installed version
+  installed_phantomjs_version <- phantomjs_version()
+
+  # Check if the installed version is latest compared to requested version.
+  # 1 indicates the installed version is newer
+  # 0 indicates the version difference is the same
+  # -1 indicates the installed version is older than requested_version.
+  utils::compareVersion(installed_phantomjs_version, requested_version) >= 0
+}
+
+
 #' Install PhantomJS
 #'
 #' Download the zip package, unzip it, and copy the executable to a system
```

Comment:
It would be good to have a paragraph that says that phantomjs is already installed and the version is >= to the requested version, this function will not try to install, unless `force` is `TRUE`.